core/src/main/java/io/questdb/cairo/wal/CheckWalTransactionsJob.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.wal;
+
+import io.questdb.cairo.CairoEngine;
+import io.questdb.cairo.PartitionBy;
+import io.questdb.cairo.TableUtils;
+import io.questdb.cairo.TxReader;
+import io.questdb.cairo.wal.seq.TableSequencerAPI;
+import io.questdb.mp.SynchronizedJob;
+import io.questdb.std.Chars;
+import io.questdb.std.str.Path;
+
+public class CheckWalTransactionsJob extends SynchronizedJob {
+    private final CairoEngine engine;
+    private final TxReader txReader;
+    private final CharSequence dbRoot;
+    private long lastProcessed = 0;
+    private final TableSequencerAPI.RegisteredTable callback = this::checkNotifyOutstandingTxnInWal;
+
+    public CheckWalTransactionsJob(CairoEngine engine) {
+        this.engine = engine;
+        txReader = new TxReader(engine.getConfiguration().getFilesFacade());
+        dbRoot = engine.getConfiguration().getRoot();
+    }
+
+    public void checkMissingWalTransactions() {
+        engine.getTableSequencerAPI().forAllWalTables(callback);
+    }
+
+    public void checkNotifyOutstandingTxnInWal(int tableId, CharSequence tableName, long txn) {
+        // todo: too much GC
```

Comment:
This one is simple to address: it's enough to convert `WalTxnNotificationTask#tableName` to `StringSink`, so that we copy table name `CharSequence` into each task. Note: this will come at the cost of additional memory & objects since `walTxnNotificationQueue`'s capacity is `4096` by default.

Reply:
This is heavily refactored in branch to remove table, let's not fix it now, job is disabled by default

Reply:
Makes sense.